src/Optimizers/BaseOptimizer.php

```diff
@@ -34,10 +96,66 @@ public function setOptions(array $options = [])
         return $this;
     }
 
+    public function useLogger(LoggerInterface $log)
+    {
+        $this->logger = $log;
+
+        return $this;
+    }
+
+
+    /**
+     * Get Binary path list
+     *
+     * @return array
+     */
+    public function binaryPathList(){
+        return $this->binaryPathList;
+    }
+
+    /**
+     * Authomatically detect the path where the image optimizes is installed
+     *
+     * @return $this
+     */
+    public function detectBinaryPath(){
+        // first check if comman is executed in a global environment
```

Comment:
This seems far to heavy. Can't we just check if the binary exists at the given path.